#==================================================================================================
# From 0af757ef2c7de5343649a0149b8b4d3488d44372 Mon Sep 17 00:00:00 2001
# From: Hui Zhou
# Date: Wed, 16 Sep 2026 10:07:53 -0500
# Subject: [PATCH] mpi.h: remove LONG_DOUBLE autoconf macros
#
# I think it is pretty safe to assume all C compilers support the "long
# double" at this point. We can discuss it again if there are issues
# spotted in the wild.
#
# PR: https://github.com/pmodels/mpich/pull/7975
#==================================================================================================
--- src/include/mpi.h.in
+++ src/include/mpi.h.in
@@ -424,9 +424,7 @@ static const MPI_Datatype mpich_mpi_long               MPICH_ATTR_TYPE_TAG(long)
 static const MPI_Datatype mpich_mpi_unsigned_long      MPICH_ATTR_TYPE_TAG(unsigned long)      = MPI_UNSIGNED_LONG;
 static const MPI_Datatype mpich_mpi_float              MPICH_ATTR_TYPE_TAG(float)              = MPI_FLOAT;
 static const MPI_Datatype mpich_mpi_double             MPICH_ATTR_TYPE_TAG(double)             = MPI_DOUBLE;
-#if @MPI_LONG_DOUBLE@ != 0x0c000000
 static const MPI_Datatype mpich_mpi_long_double        MPICH_ATTR_TYPE_TAG(long double)        = MPI_LONG_DOUBLE;
-#endif
 static const MPI_Datatype mpich_mpi_long_long_int      MPICH_ATTR_TYPE_TAG(long long int)      = MPI_LONG_LONG_INT;
 static const MPI_Datatype mpich_mpi_unsigned_long_long MPICH_ATTR_TYPE_TAG(unsigned long long) = MPI_UNSIGNED_LONG_LONG;
 struct mpich_struct_mpi_float_int       { float f; int i; };
@@ -434,9 +432,7 @@ struct mpich_struct_mpi_double_int      { double d; int i; };
 struct mpich_struct_mpi_long_int        { long l; int i; };
 struct mpich_struct_mpi_short_int       { short s; int i; };
 struct mpich_struct_mpi_2int            { int i1; int i2; };
-#if @MPI_LONG_DOUBLE_INT@ != 0x0c000000
 struct mpich_struct_mpi_long_double_int { long double ld; int i; };
-#endif
 
 static const MPI_Datatype mpich_mpi_float_int       MPICH_ATTR_TYPE_TAG_LAYOUT_COMPATIBLE(struct mpich_struct_mpi_float_int)       = MPI_FLOAT_INT;
 static const MPI_Datatype mpich_mpi_double_int      MPICH_ATTR_TYPE_TAG_LAYOUT_COMPATIBLE(struct mpich_struct_mpi_double_int)      = MPI_DOUBLE_INT;
@@ -457,9 +453,7 @@ static const MPI_Datatype mpich_mpi_short_int       MPICH_ATTR_TYPE_TAG_LAYOUT_C
 /* static const MPI_Datatype mpich_mpi_2int            MPICH_ATTR_TYPE_TAG_LAYOUT_COMPATIBLE(struct mpich_struct_mpi_2int)            = MPI_2INT
  */
 
-#if @MPI_LONG_DOUBLE_INT@ != 0x0c000000
 static const MPI_Datatype mpich_mpi_long_double_int MPICH_ATTR_TYPE_TAG_LAYOUT_COMPATIBLE(struct mpich_struct_mpi_long_double_int) = MPI_LONG_DOUBLE_INT;
-#endif
 
 static const MPI_Datatype mpich_mpi_int8_t   MPICH_ATTR_TYPE_TAG_STDINT(int8_t)   = MPI_INT8_T;
 static const MPI_Datatype mpich_mpi_int16_t  MPICH_ATTR_TYPE_TAG_STDINT(int16_t)  = MPI_INT16_T;
@@ -473,9 +467,7 @@ static const MPI_Datatype mpich_mpi_uint64_t MPICH_ATTR_TYPE_TAG_STDINT(uint64_t
 static const MPI_Datatype mpich_mpi_c_bool                MPICH_ATTR_TYPE_TAG_C99(_Bool)           = MPI_C_BOOL;
 static const MPI_Datatype mpich_mpi_c_float_complex       MPICH_ATTR_TYPE_TAG_C99(float _Complex)  = MPI_C_FLOAT_COMPLEX;
 static const MPI_Datatype mpich_mpi_c_double_complex      MPICH_ATTR_TYPE_TAG_C99(double _Complex) = MPI_C_DOUBLE_COMPLEX;
-#if @MPI_C_LONG_DOUBLE_COMPLEX@ != 0x0c000000
 static const MPI_Datatype mpich_mpi_c_long_double_complex MPICH_ATTR_TYPE_TAG_C99(long double _Complex) = MPI_C_LONG_DOUBLE_COMPLEX;
-#endif
 
 static const MPI_Datatype mpich_mpi_aint   MPICH_ATTR_TYPE_TAG(MPI_Aint)   = MPI_AINT;
 static const MPI_Datatype mpich_mpi_offset MPICH_ATTR_TYPE_TAG(MPI_Offset) = MPI_OFFSET;
